rust/numaflow-monitor/src/runtime_errors.rs

```diff
@@ -0,0 +1,130 @@
+use axum::{response::IntoResponse, Json};
+use http::StatusCode;
+use serde::{Deserialize, Serialize};
+use std::{fs, path::Path};
+use tracing::error;
+
+use crate::{
+    config::info::RuntimeInfoConfig,
+    error::{Error, ErrorRes},
+};
+
+#[derive(serde::Serialize)]
+struct ApiResponse {
+    errors: Vec<RuntimeErrorEntry>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct RuntimeErrorEntry {
+    pub container_name: String,
+    pub timestamp: String,
+    pub code: String,
+    pub message: String,
+    pub details: String,
+}
+
+/*
+File Structure for application-errors
+
+Root: /var/numaflow/runtime/
+                    └── application-errors
+                        └── udsource/
+                                ├── ts1.json
+                                └── ts2.json
+                        └── udsink/
+                                ├── ts3.json
+                                └── ts4.json
+
+*/
+pub async fn handle_runtime_app_errors() -> impl IntoResponse {
+    let runtime_info_config = RuntimeInfoConfig::default();
+    let app_err_path = Path::new(runtime_info_config.app_error_path.as_str());
```

Comment:
These things can be moved into the state https://docs.rs/axum/latest/axum/extract/struct.State.html
Probably `errors` Vec too (behind `Arc<Mutex<T>>>`).
We need to make sure the `errors` Vec doesn't grow forever.
Also, since the contents of a file doesn't ever change, we can skip reading it if its already in memory (list files in the directory, skip reading files that are in-memory).